activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -273,12 +273,18 @@ def load_target
         if find_target?
           @target = merge_target_lists(find_target, target)
         elsif target.empty? && set_through_target_for_new_record?
-          @target = if through_reflection.collection?
-            through_association.target.flat_map do |record|
-              record.association(reflection.source_reflection_name).target
+          reflections = reflection.chain
+          reflections.pop
+          reflections.reverse!
+
+          @target = reflections.reduce(through_association.target) do |middle_target, reflection|
+            if middle_target.empty?
+              break []
+            elsif reflection.collection?
+              middle_target.flat_map { |record| record.association(reflection.source_reflection_name).target }
+            else
+              middle_target.association(reflection.source_reflection_name).target
```

Comment:
In the reduce, this line will return a target that might be just one record (not a collection). If the reduce is still looping, next time middle_target will not be able to respond to empty? and will break